src/main/java/uk/gov/pay/ledger/eventpublisher/EventPublisher.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.pay.ledger.eventpublisher;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+
+import javax.inject.Inject;
+
+public class EventPublisher {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventPublisher.class);
+
+    private final SnsClient snsClient;
+    private final TopicNameArnMapper topicNameArnMapper;
+
+    @Inject
+    public EventPublisher(SnsClient snsClient, TopicNameArnMapper topicNameArnMapper) {
+        this.snsClient = snsClient;
+        this.topicNameArnMapper = topicNameArnMapper;
+    }
+
+    public void publishMessageToTopic(String message, TopicName topicName) throws EventPublisherException{
+        var topicArn = topicNameArnMapper.getArnForTopicName(topicName);
+        try {
+            PublishRequest request = PublishRequest.builder()
+                    .message(message)
+                    .topicArn(topicArn)
+                    .build();
+            snsClient.publish(request);
+        } catch (Exception e) {
+            LOGGER.warn("Failed to publish message to SNS");
```

Comment:
We should probably have some indication of what the message is.

Reply:
yes, but at this point the message is just a string, which we obvs don't want to log. Could explicitly pass in message id in to method for logging purposes?

Reply:
having discussed, agreed that responsibility for logging should be in caller, EventPublisher just needs to throw sufficiently informative exception